PHPLIB-1710: Remove dead local `.evergreen` config

paulinevos opened the pull request:
As it's covered by the `drivers-evergreen-tools` submodule now.

paulinevos committed:
Remove `.evergreen/ocsp` directory

As it seems to be covered by the `drivers-evergreen-tools` submodule,
which is loaded in evergreen functions and GH actions (added in PR
`#1584`)

jmikola commented:
I think there are some additional changes that can be applied and tested:

 * `.evergreen/config/functions.yml` refers to the local `x509gen` directory several times. Those references can be changed to `tests/drivers-evergreen-tools/.evergreen/x509gen`.
 * `.evergreen/orchestration` only contains a single `basic.json` file for a sharded cluster, and that doesn't appear to be referenced anywhere.
 * `.evergreen/auth_aws` doesn't appear to be referenced.

paulinevos commented:
@jmikola What I don't understand re: the `x509gen` references is that they're from the same exact commit as references to `x509gen` in the submodule a few lines later. That makes me feel like maybe there's a reason that points to the local ones

paulinevos committed:
Clean up `.evergreen/orchestration` directory

Its contents don't seem to be used

paulinevos committed:
Clean up `.evergreen/auth_aws`

It doesn't appear to be referenced locally anymore, only internally in the
drivers-evergreen-tools submodule

paulinevos commented:
@jmikola Oh actually, they aren't references to the local one at all:
```
cd ${DRIVERS_TOOLS}/.evergreen/csfle
```

I'll see if they're referenced anywhere else and otherwise remove

paulinevos committed:
Clean up`.evergreen/x509gen` directory

There's no need for it locally as it's covered by
the `drivers-everygreen-tools` submodule

jmikola commented on .github/workflows/tests.yml:
```diff
@@ -70,7 +70,7 @@ jobs:
           python-version: '3.13'
 
-      - name: "Setup MongoDB"
+      - name: "Set up MongoDB"
```
Was this change intentional? The old name was consistent with [PHPC](https://github.com/mongodb/mongo-php-driver/blob/v2.x/.github/workflows/tests.yml#L60). I don't feel strongly either way, just wanted to point that out.

paulinevos replied:
It was, I just changed it because of grammatical correctness, as "setup" is a noun and "to set up" is the verb. It's just a peeve I have 🫣

jmikola reviewed (approved):
Thanks, I missed the `cd` command in `functions.yml`.

Changes LGTM. I'll defer to you on the `tests.yml` change.

And please revise the PR title before squash-merging, since the PR addresses more than just OCSP.
